syft/workers/base.py

```diff
@@ -761,7 +791,13 @@ def request_remote_tensor_shape(
         shape = self.send_msg(messaging.GetShapeMessage(pointer), location=pointer.location)
         return sy.hook.create_shape(shape)
 
-    def fetch_plan(self, plan_id: Union[str, int]) -> "Plan":  # noqa: F821
+    def fetch_plan(self, plan_id):
+        if plan_id in self._objects:
+            candidate = self._objects[plan_id]
+            if isinstance(candidate, sy.Plan):
+                return candidate.copy_to_worker(sy.hook.local_worker)
```

Comment:
it would be good to explicit what happens when the if statements are False. That is, to add a return None in the end (if this is the desired behaviour)

Reply:
Indeed. Done!